internal/provider/none/installConfig.go

```diff
@@ -14,20 +14,17 @@ func (p noneProvider) AddPlatformToInstallConfig(cfg *installcfg.InstallerConfig
 		None: &installcfg.PlatformNone{},
 	}
 
-	bootstrapCidr := network.GetPrimaryMachineCidrForUserManagedNetwork(cluster, p.Log)
-	if bootstrapCidr != "" {
-		p.Log.Infof("None-Platform or SNO: Selected bootstrap machine network CIDR %s for cluster %s", bootstrapCidr, cluster.ID.String())
-		machineNetwork := []installcfg.MachineNetwork{}
-		cluster.MachineNetworks = network.GetMachineNetworksFromBoostrapHost(cluster, p.Log)
-		for _, net := range cluster.MachineNetworks {
-			machineNetwork = append(machineNetwork, installcfg.MachineNetwork{Cidr: string(net.Cidr)})
-		}
-		cfg.Networking.MachineNetwork = machineNetwork
-		cfg.Networking.NetworkType = swag.StringValue(cluster.NetworkType)
+	machineNetwork := provider.GetMachineNetworkForUserManagedNetworking(p.Log, cluster)
```

Comment:
it looks totally same code for vsphere and none, maybe we can all this in one func? 
provider.SetMachineNetworkAndTypeForUserManagedNetworking ?

Reply:
provide cfg to it?

Reply:
I didn't want to do a function that update the `cfg` in place so I returned the values and updated cfg there

Reply:
Move common code to `provider/base.go`